Userland/Libraries/LibCore/Stream.h

```diff
@@ -24,6 +24,48 @@
 
 namespace Core::Stream {
 
+template<DerivedFrom<Core::Stream::Stream> T>
+class Handle {
+public:
+    template<DerivedFrom<T> U>
+    Handle(NonnullOwnPtr<U> handle)
+        : m_handle(adopt_own<T>(*handle.leak_ptr()))
+    {
+    }
+
+    // This is made `explicit` to not accidentally create a non-owning Handle,
+    // which may not always be intended.
+    explicit Handle(T& handle)
+        : m_handle(&handle)
+    {
+    }
+
+    T* ptr()
+    {
+        if (m_handle.template has<T*>())
+            return m_handle.template get<T*>();
+        else
+            return m_handle.template get<NonnullOwnPtr<T>>();
+    }
+
+    T const* ptr() const
+    {
+        if (m_handle.template has<T*>())
+            return m_handle.template get<T*>();
+        else
+            return m_handle.template get<NonnullOwnPtr<T>>();
+    }
```

Comment:
Why would these hand out pointers and not references?

Reply:
`operator->` needs to return a pointer, so I decided to make it return a pointer. I don't think there was a deeper thought behind it.

Reply:
ok, fair enough :)

Reply:
The fact that Variant doesn't support storing `T&` and that implementing everything via `T& stream()` would require a significantly higher amount of special characters might have contributed as well.

That said, I don't think there is anything that is keeping us from making those internal helpers, in case that's what you are concerned about.